openlibrary/core/vendors.py

```diff
@@ -11,12 +18,176 @@
 
+
+def setup(config):
+    global config_amz_api, amazon_api
+    config_amz_api = config.get('amazon_api')
+    try:
+        amazon_api = AmazonAPI(
+            config_amz_api.key, config_amz_api.secret,
+            config_amz_api.id, throttling=0.9)
+    except AttributeError:
+        amazon_api = None
+
+
+class AmazonAPI:
+    """Amazon Product Advertising API 5.0 wrapper for Python
+    Creates an instance containing your API credentials.
+
+    Args:
+        key (string): Your API key.
+        secret (string): Your API secret.
+        tag (string): The tag you want to use for the URL.
+        country (string): Country code.
+        throttling (float, optional): Reduce this value to wait longer
+          between API calls.
+    """
+    # Hack: pulls all resource types from GetItemsResource
+    RESOURCES = [getattr(GetItemsResource, v) for v in
+                 vars(GetItemsResource).keys() if v.isupper()]
+
+    def __init__(self, key, secret, tag, host='webservices.amazon.com',
+                 region='us-east-1', throttling=0.9):
+        self.tag = tag
+        self.host = host
+        self.region = region
+        self.throttling = throttling
+        self.last_query_time = time.time()
+
+        self.api = DefaultApi(
+            access_key=key,
+            secret_key=secret,
+            host=host,
+            region=self.region)
+
+    def get_product(self, asin, serialize=False, **kwargs):
+        products = self.get_products([asin], **kwargs)
+        if products:
+            return next(self.serialize(p) if serialize else p for p in products)
+
+    def get_products(self, asins, serialize=False, marketplace='www.amazon.com',
+                     resources=None, **kwargs):
+        """
+        :param asins (string): One or more ItemIds like ASIN that
+        uniquely identify an item or product URL. (Max 10) Seperated
+        by comma or as a list.
+        """
+        item_ids = asins if type(asins) is list else [asins]
+
+        # Wait before doing the request
+        wait_time = 1 / self.throttling - (time.time() - self.last_query_time)
+        if wait_time > 0:
+            time.sleep(wait_time)
+        self.last_query_time = time.time()
+
+        try:
+            request = GetItemsRequest(partner_tag=self.tag,
+                                      partner_type=PartnerType.ASSOCIATES,
+                                      marketplace=marketplace,
+                                      item_ids=item_ids,
+                                      resources=resources or self.RESOURCES,
+                                      **kwargs)
+            response = self.api.get_items(request)
+            products = response.items_result.items
+            return (products if not serialize else
+                    [self.serialize(p) for p in products])
+        except Exception as exception:
+            raise exception
+
+    @staticmethod
+    def serialize(product):
```

Comment:
Is this the equivalent to `_serialize_amazon_product`? Why the name change and relocation in the module? It makes comparing old with new unnecessarily difficult.